packages/flame/lib/src/components/timer_component.dart

```diff
@@ -21,15 +23,23 @@ class TimerComponent extends Component {
   /// This overrides the [onTick] method
   /// [tickWhenLoaded] When true, will call [onTick] when the component is first
   /// loaded (default is false).
+  /// [tickCount] The number of time the timer will tick before stopping.
+  /// This is is only used when [repeat] is true. If null,
+  /// the timer will run indefinitely.
   TimerComponent({
     required double period,
     bool repeat = false,
     bool autoStart = true,
     this.removeOnFinish = false,
     VoidCallback? onTick,
     this.tickWhenLoaded = false,
+    this.tickCount,
```

Comment:
I would probably name it `tickLimit` or something like that which indicates that it is controlling when it should stop.